qcengine/programs/qchem.py

```diff
@@ -0,0 +1,254 @@
+"""
+Calls the Q-Chem executable.
+"""
+
+import os
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
+
+from qcelemental.models import Result
+from qcelemental.util import parse_version, safe_version, which
+
+from ..exceptions import InputError, UnknownError
+from ..util import disk_files, execute, popen, temporary_directory
+from .model import ProgramHarness
+
+
+class QChemHarness(ProgramHarness):
+    _defaults: Dict[str, Any] = {
+        "name": "QChem",
+        "scratch": True,
+        "thread_safe": False,
+        "thread_parallel": True,
+        "node_parallel": False,
+        "managed_memory": True,
+    }
+    version_cache: Dict[str, str] = {}
+
+    def found(self, raise_error: bool = False) -> bool:
+        return which('qchem',
+                     return_bool=True,
+                     raise_error=raise_error,
+                     raise_msg='Please install by visiting the Q-Chem website.')
+
+    def _get_qc_path(self, config: Optional['JobConfig'] = None):
+        paths = os.environ.copy()
+        paths["QCSCRATCH"] = "/tmp"
```

Comment:
call gettempdir so not platform-dep? https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir